Library/Homebrew/cmd/update.sh

```diff
@@ -45,6 +45,18 @@ git_init_if_necessary() {
     trap - EXIT
   fi
 
+  if [[ -n "${HOMEBREW_MACOS_VERSION}" ]] &&
+    [[ -z "$(git config --get homebrew.flavour 2>/dev/null)" ]]
+  then
+    if [[ "${HOMEBREW_PREFIX}" = "${HOMEBREW_MACOS_ARM_DEFAULT_PREFIX}" ]]; then
+      git config --replace-all homebrew.flavour "arm64" 2>/dev/null
+    elif [[ "${HOMEBREW_PREFIX}" = "${HOMEBREW_DEFAULT_PREFIX}" ]]; then
+      git config --replace-all homebrew.flavour "x86_64" 2>/dev/null
+    else
+      git config --replace-all homebrew.flavour "$(uname -m)" 2>/dev/null
+    fi
+  fi
```

Comment:
- feels like this should be in `brew.sh` (or even `install.rb`) rather than `update.sh`
- would be good to set a variable inside the `if`/`elif` to DRY up the repeated `git config --replace-all homebrew.flavour`
- feels like this should be inferred based on previously/newly installed formulae rather than the current architecture on existing installs (to avoid a single Rosetta/non-Rosetta run setting these incorrectly)